verilog/tools/ls/autoexpand.cc

```diff
@@ -1369,6 +1369,43 @@ std::optional<absl::string_view> AutoExpander::FindSpanToReplace(
   return replaced_span;
 }
 
+// Returns an iterator pointing at the next expansion not overlapping with *it
+auto GetNextNonOverlappingExpansion(
+    std::vector<AutoExpander::Expansion>::const_iterator it, decltype(it) end) {
```

Comment:
I'd probably rename `it`  to `start`, that might make the code below read more naturally.

Can we reformulate this whole function as a template, so that we don't have to use the awkward decltype() stuff ?

Reply:
Is a template less awkward? I think I'd rather just use the explicit type then.

Reply:
Yeah,but now we describe the same type in three different ways. `auto` for the return type, `std::vector<AutoExpander::Expansion>::const_iterator` and `decltype(it)`.

I think there is a value in readability if the same things read the same.

Reply:
Good point, changed it to templates, added a comment to make it better once we switch to C++20